From c39e6f75b87bc2188d758655512c5ff0ecaf7cb5 Mon Sep 17 00:00:00 2001
From: Xianzhong <xianzhong.li@nxp.com>
Date: Thu, 9 Aug 2018 17:29:03 +0800
Subject: [PATCH 4318/5242] MGS-4123 gpu: dts: fix axi bus error on 6dl
 sabreauto board

commit  823f4373e5281fa65aff67b9886272885bfaec62 from
https://source.codeaurora.org/external/imx/linux-imx.git

meet axi bus error when setting 3G/1G user/kernel VM space layout,
it is casued by gpu with legacy mmu, which requires cma under 2GB.

in this case, 6dl cma exceed 2G with 3G/1G user /kernel VM layerout.
Reserved memory: created CMA memory pool at 0x7a000000, size 320 MiB

set gpu base-address with 256M to avoid axi bus error on 2G board.

Signed-off-by: Xianzhong <xianzhong.li@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm/boot/dts/imx6dl.dtsi |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/boot/dts/imx6dl.dtsi b/arch/arm/boot/dts/imx6dl.dtsi
index 8adcde9..ea55ef8 100644
--- a/arch/arm/boot/dts/imx6dl.dtsi
+++ b/arch/arm/boot/dts/imx6dl.dtsi
@@ -92,7 +92,7 @@
 		gpu: gpu@130000 {
 			compatible = "fsl,imx6dl-gpu", "fsl,imx6q-gpu";
 			reg = <0x00130000 0x4000>, <0x00134000 0x4000>,
-			      <0x0 0x0>, <0x0 0x8000000>;
+			      <0x10000000 0x0>, <0x0 0x8000000>;
 			reg-names = "iobase_3d", "iobase_2d",
 				    "phys_baseaddr", "contiguous_mem";
 			interrupts = <0 9 IRQ_TYPE_LEVEL_HIGH>,
-- 
1.7.9.5

